### PR TITLE
Fixed GSL cmake interface.

### PR DESCRIPTION
**Pull request by luca-heltai**
If the gsl libraries were not in standard locations, `FindGSL.cmake` would fail to make deal.II compile. This PR fixes this by exporting the include directories both to the library itself and to user codes.

### CODE REVIEW

File: cmake/modules/FindGSL.cmake [hunk 1]
```diff
@@ -19,6 +19,7 @@
 # This module exports
 #
 #   GSL_LIBRARIES
+#   GSL_INCLUDE_LIBRARIES
```
Comment (davydden) [inline cmake/modules/FindGSL.cmake]: `GSL_INCLUDE_DIR` i suppose?

File: cmake/modules/FindGSL.cmake [hunk 2]
```diff
@@ -31,9 +32,17 @@ DEAL_II_FIND_LIBRARY(GSL_LIBRARY
...
+  )
 
 DEAL_II_PACKAGE_HANDLE(GSL
   LIBRARIES
     REQUIRED GSL_LIBRARY
-  CLEAR GSL_LIBRARY
+  INCLUDE_DIRS
+    REQUIRED GSL_INCLUDE_DIR
+  USER_INCLUDE_DIRS
+    REQUIRED GSL_INCLUDE_DIR
```
Comment (davydden) [inline cmake/modules/FindGSL.cmake]: why `REQUIRED GSL_INCLUDE_DIR` is repeated?

Comment (davydden) [inline cmake/modules/FindGSL.cmake]: oh, i see, nevermind

**Comment by luca-heltai** [comment]
Ready for re-review.

**Comment by davydden** [comment]
@luca-heltai thanks for fixing :+1:

**Pull request merged**